src/LuauRenderer/nodes/fields/renderInterpolatedStringPart.ts

```diff
@@ -2,8 +2,11 @@ import luau from "LuauAST";
 import { RenderState } from "LuauRenderer";
 
 export function renderInterpolatedStringPart(state: RenderState, node: luau.InterpolatedStringPart) {
-	// braces and newlines are escaped to be valid in luau
-	// () captures result, [] to search for any of: {}
-	// $1 fills in capture from original search
-	return node.text.replace(/([{}])/g, "\\$1").replace(/\n/g, "\\\n");
+	// escape braces and newlines, but do not touch braces within unicode escape codes
+	return node.text
+		.replace(
+			/(\\u\{[0-9A-Fa-f]+\})|([{}])/g,
```

Comment:
This regex would be cleaner with a negative lookbehind instead of capturing the Unicode sequences to like-for-like replace them. Was it a conscious decision not to use one?

Reply:
I wasn't able to find a way to do this that was readable. Open to suggestions.